examples/messenger/src/base/messenger.js

```diff
@@ -0,0 +1,162 @@
+import { map, filter } from 'rxjs/operators';
+import { SwarmClient } from '@erebos/swarm-browser';
+
+function createRandomString() {
+  return Math.random()
+    .toString(36)
+    .slice(2);
+}
+
+function decodePssEvent(data) {
+  return {
+    key: data.key,
+    data: data.msg.toObject(),
+  }
+}
+
+export default class Messanger {
+  _subscriptions = [];
+
+  client = undefined;
+  account = {};
+
+  constructor(config) {
+    this.client = new SwarmClient({ ws: config.ws });
+
+    return (async () => {
+      this.account = await this.getAccount();
+      return this;
+    })();
+  }
+
+  async getAccount() {
+    const [publicKey, overlayAddress] = await Promise.all([
+      this.client.pss.getPublicKey(),
+      this.client.pss.baseAddr(),
+    ]);
+
+    return { publicKey, overlayAddress };
+  }
+
+  async subscribe(config) {
+    if (config.onReceiveContactEvent) {
+      const subscription = await this.createContactSubscription(this.account.publicKey);
+      const contactSub = subscription.subscribe(config.onReceiveContactEvent);
+      this._subscriptions.push(contactSub);
+    }
+
+    if (config.onReceiveChatEvent && config.chats) {
+      config.chats.map(async (chat) => {
+        await this.subscribeChat(chat, config.onReceiveChatEvent);
+      });
+    }
+  }
+
+  async subscribeChat(chat, onReceiveChatEvent) {
+    const subscription = await this.createChatSubscription(chat.key, chat.topic);
+    const chatSub = subscription.subscribe(onReceiveChatEvent);
+    this._subscriptions.push(chatSub);
+  }
+
+  unsubscribe() {
+    this._subscriptions.map(s => s.unsubscribe());
+  }
+
+  async sendContactRequest(key) {
+    const [contactTopic, sharedTopic] = await Promise.all([
+      this.client.pss.stringToTopic(key),
+      this.client.pss.stringToTopic(createRandomString()),
+    ]);
+
+    await this.client.pss.setPeerPublicKey(key, contactTopic);
```

Comment:
Please also supply the address argument to `setPeerPublicKey`. 

The requesting node needs to call this again _with_ the address when response is received.